internal/resources/sample_bucket.go

```diff
@@ -0,0 +1,376 @@
+		resp.Diagnostics.AddError(
+			"Error creating sample bucket",
+			"Could not create sample bucket, unexpected error: "+errors.ErrClusterIdCannotBeEmpty.Error(),
+		)
+		return
+	}
+	var clusterId = state.ClusterId.ValueString()
+
+	if state.Id.IsNull() {
+		resp.Diagnostics.AddError(
+			"Error creating sample bucket",
+			"Could not create sample bucket, unexpected error: "+errors.ErrClusterIdCannotBeEmpty.Error(),
+		)
+		return
+	}
+	var bucketId = state.Id.ValueString()
+
+	url := fmt.Sprintf("%s/v4/organizations/%s/projects/%s/clusters/%s/sampleBuckets/%s", s.HostURL, organizationId, projectId, clusterId, bucketId)
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodDelete, SuccessStatus: http.StatusNoContent}
+	_, err := s.Client.ExecuteWithRetry(
+		ctx,
+		cfg,
+		nil,
+		s.Token,
+		nil,
+	)
+	if err != nil {
+		resourceNotFound, errString := api.CheckResourceNotFoundError(err)
+		if resourceNotFound {
+			tflog.Info(ctx, "resource doesn't exist in remote server removing resource from state file")
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError(
+			"Error Deleting the SampleBucket",
+			"Could not delete sample Bucket associated with cluster "+clusterId+": "+errString,
+		)
+		return
+	}
+}
+
+func (r *SampleBucket) validateCreateBucket(plan providerschema.SampleBucket) error {
+	if plan.OrganizationId.IsNull() {
+		return errors.ErrOrganizationIdMissing
+	}
+	if plan.ProjectId.IsNull() {
+		return errors.ErrProjectIdMissing
+	}
+	if plan.ClusterId.IsNull() {
+		return errors.ErrClusterIdMissing
+	}
+	return r.validateSampleBucketName(plan)
+}
+
+func (r *SampleBucket) validateSampleBucketName(plan providerschema.SampleBucket) error {
+	if (!plan.Name.IsNull() && !plan.Name.IsUnknown()) && !providerschema.IsTrimmed(plan.Name.ValueString()) {
+		return fmt.Errorf("name %s", errors.ErrNotTrimmed)
+	}
+
+	if !isValidSampleName(plan.Name.ValueString()) {
+		return errors.ErrInvalidSampleBucketName
+	}
+
+	return nil
+}
+
+// retrieveSampleBucket retrieves sample bucket information for a specified organization, project, cluster and sample bucket ID.
+func (s *SampleBucket) retrieveSampleBucket(ctx context.Context, organizationId, projectId, clusterId, bucketId string) (*providerschema.SampleBucket, error) {
+	url := fmt.Sprintf("%s/v4/organizations/%s/projects/%s/clusters/%s/sampleBuckets/%s", s.HostURL, organizationId, projectId, clusterId, bucketId)
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodGet, SuccessStatus: http.StatusOK}
+	response, err := s.Client.ExecuteWithRetry(
+		ctx,
+		cfg,
+		nil,
+		s.Token,
+		nil,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", errors.ErrExecutingRequest, err)
+	}
+
+	sampleBucketResp := samplebucketapi.GetSampleBucketResponse{}
+	err = json.Unmarshal(response.Body, &sampleBucketResp)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", errors.ErrUnmarshallingResponse, err)
+	}
+	sampleStats := providerschema.NewStats(*sampleBucketResp.Stats)
```

Comment:
Here, we are not handling the null pointer case

Reply:
It appears we aren't handling the null pointer case in multiple places. eg. for [backup stats](https://github.com/couchbasecloud/terraform-provider-couchbase-capella/blob/AV-70846_add_import_sample_data_set_apis/internal/resources/backup.go#L474)